programs/mango-v3-reimbursement/src/instructions/reimburse.rs

```diff
@@ -0,0 +1,124 @@
+use std::mem::size_of;
+
+use crate::state::{Group, ReimbursementAccount, Row};
+use crate::Error;
+use anchor_lang::{__private::bytemuck, prelude::*};
+use anchor_spl::token::{self, Mint, Token, TokenAccount};
+
+#[derive(Accounts)]
+#[instruction(token_index: usize)]
+pub struct Reimburse<'info> {
+    #[account (
+        constraint = group.load()?.has_reimbursement_started()
+    )]
+    pub group: AccountLoader<'info, Group>,
+
+    #[account(
+        mut,
+        address = group.load()?.vaults[token_index]
+    )]
+    pub vault: Account<'info, TokenAccount>,
+
+    #[account(
+        mut,
+        constraint = token_account.owner == mango_account_owner.key()
+    )]
+    pub token_account: Box<Account<'info, TokenAccount>>,
+    #[account(
+        mut,
+        seeds = [b"ReimbursementAccount".as_ref(), group.key().as_ref(), mango_account_owner.key().as_ref()],
+        bump,
+        constraint = group.load()?.is_testing() || !reimbursement_account.load()?.reimbursed(token_index),
+        constraint = group.load()?.is_testing() || !reimbursement_account.load()?.claim_transferred(token_index)
```

Comment:
minor: this claim_transferred check is confusing because it can never be true while !reimbursed